sopht_mpi/utils/mpi_utils_2d.py

```diff
@@ -27,14 +24,23 @@ def __init__(
         # Setup MPI environment
         self.world = MPI.COMM_WORLD
         # Automatically create topologies
-        self.rank_distribution = rank_distribution
+        if rank_distribution is None:
+            self.rank_distribution = [0] * self.grid_dim
+            self.rank_distribution[
+                -1
+            ] = 1  # to align at least one dimension for fft operations
+        else:
+            self.rank_distribution = rank_distribution
+        assert (
+            1 in self.rank_distribution
+        ), f"Rank distribution {self.rank_distribution} needs to be aligned in at least one direction for fft"
```

Comment:
Instead of the `assert`, let's raise a `ValueError` here saying invalid rank distribution provided, followed by the text you have here.

Reply:
good point. fixed in [739e691](https://github.com/fankiat/sopht-mpi/pull/2/commits/739e69182673177ae278a0eb79ac849395ff6ea5)